cmake/public/xpu.cmake

```diff
@@ -1,30 +1,79 @@
-# ---[ xpu
+# XPU backend stacks (compiler, runtime, libraries, tools)
 
-# Poor man's include guard
-if(TARGET torch::xpurt)
+# output variables:
+#
+#  PYTORCH_FOUND_XPU
+#  -- The flag to indicate whether XPU backend stacks are setup successfully or not.
+#
+#  PYTORCH_SYCL_LIBRARIES
+#  -- Full name of SYCL runtime libraries.
+
+# output INTERFACE target:
+#
+#  torch::syclrt
+#  -- Define it for linking usage, e.g. target_link_library(tgt torch::syclrt)
+#
+
+set(PYTORCH_FOUND_XPU FALSE)
+
+if(SYCL_cmake_included)
   return()
 endif()
+set(SYCL_cmake_included true)
 
-# Find SYCL library.
+# SYCL compiler and runtime setup
 find_package(SYCLToolkit REQUIRED)
-if(NOT SYCL_FOUND)
-  set(PYTORCH_FOUND_XPU FALSE)
+if(NOT SYCLTOOLKIT_FOUND)
+  message("Cannot find SYCL compiler tool kit!")
+  return()
+endif()
```

Comment:
It just prints a message string. Does it mean the build process will be ongoing If users set USE_XPU but cannot find SYCLTOOLKIT?

Reply:
Yes. The process is ongoing w/o building XPU backend staffs.